File: buildengine/testdata/projects/alphakotlin/src/main/kotlin/ftl/alpha/Alpha.kt
package ftl.alpha

import ftl.builtin.Empty
import ftl.other.Other
import xyz.block.ftl.Context
import xyz.block.ftl.Verb

data class EchoRequest(val name: String?)
data class EchoResponse(val message: String)

@Verb
fun echo(context: Context, req: EchoRequest): EchoResponse {
  val other = Other()
  return EchoResponse(message = "Hello, ${req.name ?: "anonymous"}!.")
}

File: go-runtime/goplugin/testdata/alpha/alpha.go
package alpha

import (
	"context"
	"fmt"

	"ftl/other"

	"github.com/block/ftl/go-runtime/ftl" // Import the FTL SDK.
)

type EchoRequest struct {
	Name ftl.Option[string] `json:"name"`
}

type EchoResponse struct {
	Message string `json:"message"`
}

//ftl:verb
func Echo(ctx context.Context, req EchoRequest, oc other.EchoClient) (EchoResponse, error) {
	oc(ctx, other.EchoRequest{})
	return EchoResponse{Message: fmt.Sprintf("Hello, %s!", req.Name.Default("anonymous"))}, nil
}
